core/src/test/java/org/apache/accumulo/core/data/KeyBuilderTest.java

```diff
@@ -0,0 +1,282 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.accumulo.core.data;
+
+import org.apache.hadoop.io.Text;
+import org.junit.Test;
+
+import static org.junit.Assert.*;
+
+public class KeyBuilderTest {
```

Comment:
> It would be nice to test that all builder methods fail fast when passed null.

This is probably better handled with static analysis tools and the use of `@Nonnull` annotations than unit test,, which is a out of scope for this change.